### PR TITLE
[BC] Change permission checks to `view` for `searchById`

### LINKED ISSUE #221: [BC] Change Permission check for `getById` to VIEW
When searching elements by ID the `view` permission needs to be checked.

Currently the `list` permission is checked

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request
Resolves #221 

## Additional info
Permission check for searchById is now `view` instead of `list`

### CODE REVIEW

File: src/Service/Search/SearchService/AbstractSearchHelper.php [hunk 1]
```diff
@@ -46,7 +46,8 @@ public function __construct(
     public function addSearchRestrictions(
         SearchInterface $search,
         string $userPermission,
-        string $workspaceType
+        string $workspaceType,
+        PermissionTypes $permissionType = PermissionTypes::LIST
```
Comment (martineiber) [inline src/Service/Search/SearchService/AbstractSearchHelper.php:50]: I think we can remove some Default values? 
e.g. this is used in search() which also has a default value..

**Review by martineiber — CHANGES_REQUESTED** [review]

File: src/Service/Search/SearchService/AbstractSearchHelper.php (hunk 1 above)
Comment (mattamon) [reply src/Service/Search/SearchService/AbstractSearchHelper.php:50]: Hmm, I would agree if it were e.g. private methods. But those are all public and cann be called, so I would leave the default, unless we want to force the permissionType.

Comment (martineiber) [reply src/Service/Search/SearchService/AbstractSearchHelper.php:50]: Works for me :+1:

**Review by martineiber — APPROVED** [review]
LGTM :t-rex:

**Pull request merged**